internal/plugin/connectors/tcp/mssql/connector.go

```diff
@@ -0,0 +1,111 @@
+package mssql
+
+import (
+	"context"
+	"fmt"
+	"net"
+
+	"github.com/cyberark/secretless-broker/pkg/secretless/log"
+	"github.com/cyberark/secretless-broker/pkg/secretless/plugin/connector"
+
+	mssql "github.com/denisenkom/go-mssqldb"
+)
+
+// SingleUseConnector is used to create an authenticated connection to an MSSQL target
+type SingleUseConnector struct {
+	backendConn net.Conn
+	clientConn  net.Conn
+	logger      log.Logger
+}
+
+// Connect implements the tcp.Connector func signature
+//
+// It is the main method of the SingleUseConnector. It:
+//   1. Constructs connection details from the provided credentials map.
+//   2. Dials the backend using credentials.
+//   3. Runs through the connection phase steps to authenticate.
+//   4. Pipes all future bytes unaltered between client and server.
+//
+// Connect requires "user", "password", and "address" credentials.
+func (connector *SingleUseConnector) Connect(
+	clientConn net.Conn,
+	credentialValuesByID connector.CredentialValuesByID,
+) (net.Conn, error) {
+
+	connector.clientConn = clientConn
+
+	err := connector.ConnectToBackend(credentialValuesByID)
+	if err != nil {
+		connector.logger.Errorf("Failed to connect to mssql server: %s", err)
+		return nil, err
+	}
+
+	err = connector.performPreLoginHandshake()
+	if err != nil {
+		connector.logger.Errorf("Failed to handle client prelogin request: %s", err)
+		return nil, err
+	}
+
+	// Now that the backend connection is authenticated, send the client
+	// a successful authentication response.
+	if _, err = clientConn.Write(connector.CreateAuthenticationOKMessage()); err != nil {
```

Comment:
There seems to be some logic missing for sending a not-OK message to the client. I'm not sure how this connector is currently propagating errors to the client, if at all.

Reply:
oh i see that now. good catch!

Reply:
added code to send the client a generic error. we should also send the actual error but i didn't get to that